src/Microsoft.Extensions.OData.Migration/Formatters/Deserialization/ODataMigrationResourceDeserializer.cs

```diff
@@ -0,0 +1,84 @@
+﻿// ------------------------------------------------------------------------------
+// <copyright company="Microsoft Corporation">
+//     Copyright © Microsoft Corporation. All rights reserved.
+// </copyright>
+// ------------------------------------------------------------------------------
+
+namespace Microsoft.Extensions.OData.Migration.Formatters.Deserialization
+{
+    using Microsoft.AspNet.OData.Formatter.Deserialization;
+    using Microsoft.OData;
+    using Microsoft.OData.Edm;
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Linq;
+    using System;
+    using System.IO;
+    using System.Linq;
+    using System.Text;
+
+    /// <summary>
+    /// This customized deserializer for resources will modify request bodies that represent OData entities
+    /// to convert OData V3 request body conventions to V4 request body conventions (e.g. quoted longs to longs)
+    /// </summary>
+    internal class ODataMigrationResourceDeserializer : ODataResourceDeserializer
+    {
+        public ODataMigrationResourceDeserializer(ODataDeserializerProvider provider)
+            : base(provider)
+        {
+        }
+
+        /// <summary>
+        /// If the request body is V3, preempts the base deserializer by first modifying the JSON request body, then
+        /// swapping out the HTTP request stream with the modified body that can be successfully read by the base deserializer.
+        /// </summary>
+        /// <param name="messageReader">An ODataMessageReader</param>
+        /// <param name="type">Unused by this read method</param>
+        /// <param name="readContext">Context to track state and settings of deserialization</param>
+        /// <returns>Deserialized object from request body</returns>
+        public override object Read(ODataMessageReader messageReader, Type type, ODataDeserializerContext readContext)
+        {
+            IEdmTypeReference edmType = GetEdmType(readContext, type);
```

Comment:
Seems the code is similar to the one in another deserializer. Is it possible to put the common code in one place?